src/ucp/api/device/ucp_device_impl.h

```diff
@@ -60,6 +60,23 @@ UCS_F_DEVICE void ucp_device_request_init(uct_device_ep_t *device_ep,
 }
 
 
+/**
+ * Macro for device put operations with retry logic
+ */
+#define UCP_DEVICE_PUT_BLOCKING(level, uct_device_ep_put, device_ep, ...) \
```

Comment:
```suggestion
#define UCP_DEVICE_CALL_BLOCKING_UCT(_uct_device_func, _device_ep, ...) \
```
level can be part of _uct_device_func